### PR TITLE
MDEV-15789 - mysqlslap --auto-generate-sql-guid-primary uses incorrect table

### PR DESCRIPTION
**Pull request by chriscalender**
This is a fix for MDEV-15789, a minor fix to mysqlslap when using --auto-generate-sql-guid-primary.

### CODE REVIEW

**Comment by svoj** [comment]
Hi @chriscalender,

Thanks for your contribution! I think this fix is acceptable even for 5.5. Either you or I can cherry-pick this to 5.5.

Do you plan to fix `--auto-generate-sql-secondary-indexes` separately?

Could you add test case? Probably take one from mysql/mysql-server@c6ea57ebf.

Could you edit commit comment so that it says something useful? E.g. give reasons why this change was required. You should also mention JIRA task id in first line of your commit comment, like:
```
MDEV-15789 - mysqlslap --auto-generate-sql-guid-primary uses incorrect table

blah-blah-blah
```

Thanks,
Sergey

**Comment by chriscalender** [comment]
Hi @svoj,

Thank you for the feedback!

I just added the fix for --auto-generate-sql-secondary-indexes to this.  I thought it might be best to have these same two fixes in the same commit.  I did not catch that bug at first, is the only reason I did not add it.

I've not performed a cherry-pick yet.  I'm happy to do so if you let me know what I need to do for it.  From what I do know, I think the command would look something like:

git checkout mariadb-5.5 && git cherry-pick abc123...

However, perhaps this commit must be completed first?  Also, please let me know what the correct syntax would be.

As for the test case, I've not added one of those before, so I need to dig into that a bit more.

Again, thank you so much for your feedback and advice!  I greatly appreciate it, and want to learn so I can contribute much more in the future.

**Comment by svoj** [comment]
Based on current state of this PR, most probably the following should be done:

1. Prepare test case, make sure it passes with your changes and doesn't pass without your changes. Do not commit yet.

2. Reset second commit (changes should be preserved, staged for another commit):
```
git reset HEAD^
```

3. Amend changes into the first commit and write some decent commit message:
```
git commit --amend mysql-test/t/mysqlslap.test mysql-test/t/mysqlslap.result client/mysqlslap.c
```

4. Make sure your commit looks good:
```
git show
```

5. Force push
```
git push --force
```

6. Make sure this PR was updated properly (check Commits and Files Changed tabs).

7. Checkout 5.5
```
git checkout 5.5
```

8. Cherry-pick last commit from your branch (I assume branch name is MDEV-15789)
```
git cherry-pick MDEV-15789
```

Hopefully this will be done automatically, but conflicts may happen. Resolving them would add some complexity, then you may try to cherry-pick this patch manually for the first time.

9. Force push to remote MDEV-15789 branch
```
git push --force origin HEAD:MDEV-15789
```

10. Check if PR was updated properly. If so, click "edit" button next to PR title and tell that you want to have it merged to 5.5.

That should be it. I can take over some steps or help you if you experience some problems.

**Comment by chriscalender** [comment]
Hi Sergey,

Yes, I am still planing to do this.  Sorry for the delay.  I just finished
a big project today, so hopefully I can work on tomorrow, and Friday, if
need be.

Thanks so much!

Best wishes,
Chris

**Comment by chriscalender** [comment]
Hello @svoj !

I got a chance to complete most of this today.  I was able to create tests and results (step #1).

Step #2 did not work for me ("git reset HEAD^"), however, it seems I was still able to successfully complete steps #3-#7.

I then ran into a problem with the cherry-pick, step #8:

shell>git cherry-pick MDEV-15789
fatal: bad revision 'MDEV-15789'

So I think I will need your help with that.

Lastly, after re-looking everything over, I'm not sure how to edit the commit comment.

Hopefully, that is all, but please just let me know if there is anything else I can, or need to, do.  Many thanks! :)

File: mysql-test/t/mysqlslap.test [hunk 1]
```diff
@@ -80,3 +80,13 @@ DROP DATABASE bug58090;
...
 DROP TABLE t1;
+
+--echo #
+--echo # Bug MDEV-15789 (Upstream: #80329): MYSQLSLAP OPTIONS --AUTO-GENERATE-SQL-GUID-PRIMARY and --AUTO-GENERATE-SQL-SECONDARY-INDEXES DONT WORK
+--echo #
+
+--exec $MYSQL_SLAP --concurrency=1 --silent --iterations=1 --number-int-cols=2 --number-char-cols=2 --auto-generate-sql --auto-generate-sql-guid-primary --create-schema=slap
+DROP TABLE t1;
+--exec $MYSQL_SLAP --concurrency=1 --silent --iterations=1 --number-int-cols=2 --number-char-cols=2 --auto-generate-sql --auto-generate-sql-secondary-indexes=1 --create-schema=slap
+DROP TABLE t1;
```
Comment (svoj) [inline mysql-test/t/mysqlslap.test]: Why `DROP TABLE` here? FWICS there's no table `t1` created.

**Comment by svoj** [comment]
Thanks for addressing my comments. There're still a few things to improve:
- re-target for 5.5
- squash commits
- write commit comment

I can take it from here, but I won't be able to use this shiny "Squash and merge" button on github. Hope that works.

File: mysql-test/t/mysqlslap.test (hunk 1 above)
Comment (svoj) [reply mysql-test/t/mysqlslap.test]: Also MySQL does `--number-char-cols=3`, while in your patch it is `--number-char-cols=2`. Why was this change needed?

Comment (chriscalender) [reply mysql-test/t/mysqlslap.test]: I see t1 created even when the command fails (from both commands), hence the 2 DROP TABLEs.

As for --number-char-cols=2, I was just trying to make the table slightly smaller, while still demonstrating the error.  However, I'm happy to make them the same.

**Comment by chriscalender** [comment]
I updated the test to use --number-char-cols=3 and pushed.

**Comment by chriscalender** [comment]
Btw, if I pushed something to 5.5, I did not mean to (but I don't know how to remove/delete it).

**Comment by chriscalender** [comment]
Or if you prefer, I can scrap these commits, download and edit 5.5, and push that (with a proper comment).  Just let me know if that would be better at this point.

**Comment by svoj** [comment]
Pushed to 5.5 d9b159a2027c56c5c87385cfe1ae43b8c73a97b6 with a few minor modifications. I also had to remove `DROP TABLE t1`, because otherwise I was getting an error as I expected:
```
mysqltest: At line 89: query 'DROP TABLE t1' failed: 1051: Unknown table 't1'
```

Thanks!

**Pull request closed without merging**